File: snippets/java/src/main/java/foundations/actions/NotificationService.java
package foundations.actions;

import dev.restate.sdk.annotation.Handler;
import dev.restate.sdk.annotation.Service;

@Service
public class NotificationService {
  @Handler
  public void sendEmail(EmailRequest req) {
    // Implementation
  }

  @Handler
  public void sendNotification(String msg) {
    // Implementation
  }

  public record EmailRequest(String userId, String message) {}
}

File: snippets/python/src/foundations/actions/notification_service.py
from typing import Dict

import restate

notification_service = restate.Service("NotificationService")


@notification_service.handler()
async def send_email(ctx: restate.Context, req: Dict[str, str]) -> None:
    pass


@notification_service.handler()
async def send_reminder(ctx: restate.Context, req: Dict[str, str]) -> None:
    pass
